lib/controller/response.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Controller
+  class Response
+    def initialize(controller, result, options = {})
+      @controller = controller
+      @result = result
+      @options = options
+    end
+
+    def execute
+      if result.errors.blank? && result.resource.blank?
+        controller.head result.status
+      elsif result.errors.any?
```

Comment:
Decided to be conservative and return errors if there are any instead of a resource.